t/data_migration_pipeline/main.go

```diff
@@ -52,37 +58,57 @@ func parseDataModel(record []string) *DataModel {
 }
 
 // emitResult emits data models to be written to Spanner
-func emitResult(s beam.Scope, lines beam.PCollection) beam.PCollection {
+func emitResult(ctx context.Context, s beam.Scope, lines beam.PCollection) beam.PCollection {
 	dataModels := beam.ParDo(s, func(line string, emit func(*DataModel)) {
 		reader := csv.NewReader(strings.NewReader(line))
 		csvLine, err := reader.Read()
 		if err != nil {
 			log.Fatalf("Failed to read record: %v", err)
 		}
 		emit(parseDataModel(csvLine))
+		count.Inc(ctx, 1)
 	}, lines)
+
 	return dataModels
 }
 
 func main() {
+	// Handle context cancellation.
+	ctx, cancel := signal.NotifyContext(context.Background(),
+		syscall.SIGINT, syscall.SIGTERM)
+	defer cancel()
+
+	// Set up metric
+	ctx = metrics.SetPTransformID(ctx, "emitResult")
```

Comment:
Would recommend using the `realMain` pattern. E.g. https://github.com/abcxyz/github-metrics-aggregator/blob/7e8df8db3fa0dbe69b9a7833f8205002a8d9b66a/pipeline/leech/main.go#L53